pkg/controller/core/resourceflavor_controller.go

```diff
@@ -96,21 +141,70 @@ func (r *ResourceFlavorReconciler) Update(e event.UpdateEvent) bool {
 	log := r.log.WithValues("resourceFlavor", klog.KObj(flv))
 	log.V(2).Info("ResourceFlavor update event")
 
+	if flv.DeletionTimestamp != nil {
+		return true
+	}
+
 	if cqNames := r.cache.AddOrUpdateResourceFlavor(flv.DeepCopy()); len(cqNames) > 0 {
 		r.qManager.QueueInadmissibleWorkloads(cqNames)
 	}
 	return false
 }
 
+func (r *ResourceFlavorReconciler) NotifyClusterQueueUpdate(cq *kueue.ClusterQueue) {
+	r.cqUpdateCh <- event.GenericEvent{Object: cq}
+}
+
 func (r *ResourceFlavorReconciler) Generic(e event.GenericEvent) bool {
-	r.log.V(3).Info("Ignore generic event", "obj", klog.KObj(e.Object), "kind", e.Object.GetObjectKind().GroupVersionKind())
-	return false
+	r.log.V(2).Info("Got generic event", "obj", klog.KObj(e.Object), "kind", e.Object.GetObjectKind().GroupVersionKind())
+	return true
+}
+
+// cqHandler signals the controller to reconcile the resourceFlavor
+// associated to the clusterQueue in the event.
+// Since the events come from a channel Source, only the Generic handler will
+// receive events.
+type cqHandler struct {
+	cache *cache.Cache
+}
+
+func (h *cqHandler) Create(event.CreateEvent, workqueue.RateLimitingInterface) {
+}
+
+func (h *cqHandler) Update(event.UpdateEvent, workqueue.RateLimitingInterface) {
+}
+
+func (h *cqHandler) Delete(event.DeleteEvent, workqueue.RateLimitingInterface) {
+}
+
+func (h *cqHandler) Generic(e event.GenericEvent, q workqueue.RateLimitingInterface) {
```

Comment:
Add a comment that we are checking the flavor for the previous version of the object, which might now be free